pkg/kubeapiserver/options/authentication_test.go

```diff
@@ -1051,9 +1075,13 @@ jwt:
 			}
 
 			resultConfig, err := opts.ToAuthenticationConfig()
-			if err != nil {
-				t.Fatal(err)
+			if (err != nil) != (testcase.expectErr != "") {
+				t.Fatalf("Got err: %v; Want err: %v", err, testcase.expectErr)
```

Comment:
While I understand what the check does (ensure that the latter shall be defined if the former happened (and vice versa)), I think the message is a bit confusing. AFAIU, this checks that the tests is properly configured. This is different from the error below which ensures that the proper error message has been emitted.

```suggestion
			if (err != nil) != (testcase.expectErr != "") {
				t.Fatalf("Test configuration mismatch; Got err: %v; Want err: %v", err, testcase.expectErr)
```